src/main/java/frc/robot/subsystems/BallPath/Shooter/PIDShooterTrackingImpl.java

```diff
@@ -0,0 +1,363 @@
+                if(turretEncoderReadingPosition < rightLimit - turretBuffer && canSeeTarget == 0.0){
+                    turretMotor.set(ControlMode.PercentOutput, turretSpeed);
+                }else{
+                    flipRight = false;
+                }
+            // if we are at the right limit, flip to the left
+            }else if(turretEncoderReadingPosition > rightLimit || flipLeft == true){
+                turretReady = false;
+                flipLeft = true;
+                if(turretEncoderReadingPosition > leftLimit + turretBuffer && canSeeTarget == 0.0){
+                    turretMotor.set(ControlMode.PercentOutput, -turretSpeed);
+                }else{
+                    flipLeft = false;
+                }
+            }    
+        // if we want to shoot the fender shot, set the turret position to be 0 (straight)
+        // this may change if we have a turret that can turn 180 degrees
+        }else{
+            if(turretHoodPosition >= setPointHood - hoodBuffer && turretHoodPosition <= setPointHood + hoodBuffer){
+                hoodMotor.set(ControlMode.PercentOutput, 0);
+                hoodReady = true;
+            }else if(turretHoodPosition <= setPointHood - hoodBuffer){
+                hoodMotor.set(ControlMode.PercentOutput, hoodSpeed);
+                hoodReady = false;
+            }else if (turretHoodPosition >= setPointHood + hoodBuffer){
+                hoodMotor.set(ControlMode.PercentOutput, -hoodSpeed);
+                hoodReady = false;
+            }
+        }
+        
+        // setpoint for the shooter wheel
+
+        if(setPointShooterPID != 0){
+            currentOutput = shooterPid.calculate(shooterEncoderReadingVelocity, setPointShooterPID);
+            currentOutput += 0.01; // hack "feed forward"
+            currentOutput = Utils.normalizePwm(currentOutput);
+        } else {
+            currentOutput = 0;
+        }
+        this.shooterMotor.set(ControlMode.PercentOutput, currentOutput);
+    }
+
+    @Override
+    public void findAndCenterTarget() {}
+
+    @Override
+    public void centerTarget(double tx){}
+
+    @Override
+    public void getDistance(double ty, double angle1, double angle2){}
+
+    @Override
+    public boolean blocking() {
+        boolean spinningUp = setPointShooterPID > 0 && !shooterPid.atSetpoint();
+        // threshold to allow for a little bit of sensor movement around 0, not requiring absolute stillness
+        double threshold = 500;
+        double absoluteWheelSpeed = Math.abs(shooterEncoderReadingVelocity);
+        boolean spinningDown = setPointShooterPID == 0 && absoluteWheelSpeed > threshold;
+        return spinningUp || spinningDown;
+        // return false;
+    }
+
+    @Override
+    public boolean readyToShoot(){
+        boolean shooterReady = false;
+        if(Math.abs(shooterEncoderReadingVelocity) > setPointShooterPID - 500 && Math.abs(shooterEncoderReadingVelocity) < shooterEncoderReadingVelocity + 500){
```

Comment:
Same here as in the other file re. using the PID controller's tolerances and `atSetpoint()` method